main.go

```diff
@@ -215,7 +215,9 @@ func main() {
 	http.Handle("/metrics", prometheus.Handler())
 	http.HandleFunc("/probe",
 		func(w http.ResponseWriter, r *http.Request) {
+			sc.RLock()
 			probeHandler(w, r, sc.C)
+			sc.RUnlock()
```

Comment:
This will hold the lock for the entire duration of a probe, which could be a while. Take the lock only to copy the pointer to a local variable.